lang/test/org/partiql/lang/planner/memorydb/QueryEngine.kt

```diff
@@ -0,0 +1,211 @@
+    var enableDebugOutput = false
+
+    /** Given a [BindingName], inform the planner the unique identifier of the global variable (usually a table). */
+    private val globalVariableResolver = GlobalVariableResolver { bindingName ->
+        // The planner has asked us to resolve a global variable named [bindingName]. let's do so and return the
+        // UUID of the table.  This will get packaged into a (global_id <uuid>) node (a reference to an
+        // unambiguously global variable).
+        db.findTableMetadata(bindingName)?.let { tableMetadata ->
+            GlobalResolutionResult.GlobalVariable(tableMetadata.tableId.toString())
+        } ?: GlobalResolutionResult.Undefined
+    }
+
+    /** Given a global variable's unique id, informs the planner about the static type (schema) of the global variable. */
+    private val staticTypeResolver = StaticTypeResolver { uniqueId ->
+        val tableMetadata = db.getTableMetadata(UUID.fromString(uniqueId))
+        // Tables are a bag of structs.
+        // TODO: at some point we'll populate this with complete schema information.
+        BagType(
+            StructType(
+                // TODO: nothing in the planner uses the fields property yet
+                fields = emptyMap(),
+                // TODO: nothing in the planner uses the contentClosed property yet
+                // But "technically" do have open content since nothing is constraining the fields in the table.
+                contentClosed = false,
+                // TODO: The FilterScanTokeyLookup pass does use this.
+                primaryKeyFields = tableMetadata.primaryKeyFields
+            )
+        )
+    }
+
+    val bindings = object : Bindings<ExprValue> {
+        /**
+         * This function is called by the `(global_id <unique_id>)` expression to fetch an [ExprValue] for a resolved
+         * global variable, which is almost always a database table.
+         */
+        override fun get(bindingName: BindingName): ExprValue {
+            // TODO: PartiQL may need some additional cleanup here because, perhaps very confusingly, the bindingName
+            // passed here contains the UUID of the table and *not* its name.  It should also always specify a
+            // case-sensitive binding name.  Really, we should reconsider if [PlannerPipeline] should use
+            // [Bindings<T>] at all, perhaps another interface that's more narrow should be used instead.
+            // Another difference in how PlannerPipeline uses Bindings<ExprValue> (and argument for using a new
+            // interface) is that the bindingName here is guaranteed to be valid because otherwise planning would have
+            // been aborted.  If the lookup fails for some reason it would mean the plan is invalid.  But with
+            // the [CompilerPipeline] these checks don't exist and therefore the [Bindings<T>] implementation is
+            // expected to throw if the variable does not exist.
```

Comment:
Does this cleanup apply to PartiQL as a whole or just the planner's use of `BindingName`? If it applies to PartiQL as a whole, it may be better to capture this comment in a GitHub issue rather than buried in a TODO of a test class.

Reply:
I should probably be more specific in this comment--it applies to only `PlannerPipeline` and not the `CompilerPipeline`.

Reply:
Will reword.